applications/ContactStructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/displacement_contact_criteria.h

```diff
@@ -181,20 +202,33 @@ class DisplacementContactCriteria
                     dof_value = it_dof->GetSolutionStepValue(0);
                     dof_incr = rDx[dof_id];
 
-                    const auto curr_var = it_dof->GetVariable();
-                    disp_solution_norm += dof_value * dof_value;
-                    disp_increase_norm += dof_incr * dof_incr;
-                    disp_dof_num++;
+                    const auto& r_curr_var = it_dof->GetVariable();
+                    if ((*p_check_disp)(r_curr_var)) {
+                        disp_solution_norm += std::pow(dof_value, 2);
+                        disp_increase_norm += std::pow(dof_incr, 2);
+                        ++disp_dof_num;
+                    } else {
+                        KRATOS_DEBUG_ERROR_IF_NOT((r_curr_var == ROTATION_X) || (r_curr_var == ROTATION_Y) || (r_curr_var == ROTATION_Z)) << "Variable must be a ROTATION and it is: " << r_curr_var.Name() << std::endl;
+                        rot_solution_norm += std::pow(dof_value, 2);
+                        rot_increase_norm += std::pow(dof_incr, 2);
+                        ++rot_dof_num;
```

Comment:
this Criteria looks pretty similar to the new `GenericCriteria` in the Core
Maybe you could use that one instead?

Reply:
There are things specific to the contact as the decomposition of the LM, so it is not general enough